internal/provider/kubernetes/httproute.go

```diff
@@ -145,6 +147,20 @@ func (r *httpRouteReconciler) Reconcile(ctx context.Context, request reconcile.R
 		routeKey := utils.NamespacedName(&route)
 		if routeKey == request.NamespacedName {
 			found = true
+			gws, err := r.validateParentRefs(ctx, &route)
+			if err != nil {
+				// Remove the route from the watchable map since it's invalid.
+				r.resources.HTTPRoutes.Delete(routeKey)
+				return reconcile.Result{}, fmt.Errorf("invalid parentRefs for httproute %s/%s: %v",
+					routeKey.Namespace, routeKey.Name, err)
+			}
+			if len(gws) == 0 {
+				// Remove the route from the watchable map since it doesn't reference
+				// a managed Gateway.
+				r.resources.HTTPRoutes.Delete(routeKey)
+				return reconcile.Result{}, nil
+			}
+			log.Info("validated httproute parentRefs")
```

Comment:
Since we're iterating over every route and adding them all to the watchable map, wouldn't we want to do this check for every route, not *just* the one that triggered the reconcile?